assets/migrations/postgres/004_new_user_parts_columns.sql

```diff
@@ -0,0 +1,114 @@
+-- +goose NO TRANSACTION
+-- This migration adds 3 more columns to the tuple and changelog tables, and fills past and future data with a trigger.
+-- This migration also adds more indexes to the tuple table. It does so concurrently so that the main table isn't locked
+
+-- +goose Up
+
+ALTER TABLE tuple
+    ADD COLUMN IF NOT EXISTS user_object_type TEXT,
+    ADD COLUMN IF NOT EXISTS user_object_id TEXT,
+    ADD COLUMN IF NOT EXISTS user_relation TEXT;
+
+-- jon becomes (..., jon, ...)
```

Comment:
The comments (here and elsewhere) describing the replacements of `...` are misleading. The current transformation function `set_new_columns` does not replace with `...`.

<img width="1471" alt="Screen Shot 2023-05-23 at 1 58 27 PM" src="https://github.com/openfga/openfga/assets/2899204/f4de05ca-c8bc-43ec-8b4e-cbce77c73c22">

Reply:
Oops this was the old approach haha will update